docs/articles/baseplatform/creating-a-new-layer.rst

```diff
@@ -180,5 +180,20 @@ and
 This can save you some later pain if the code gets moved to a new server, because in this way you
 won't need to modify all the recipes.
 
+Optionally depending on other layers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In some cases a layer might have recipes optional dependencies on other layers. For instance, one
+might want to append on a recipe for a Qt application which will only be built for some qt specific
+image. Adding the recipe directly to the layer will cause bitbake to generate errors about dangling
+appends when building non-Qt images, which would not include any Qt layers. To mitigate this, create
+a ``layers`` directory containing directories named after the layers that is required, ``qt5`` in the
+Qt application example. Then add all .bb and .bbappend files needed to that layer. Finally in
+``layer.conf`` make sure that the ``.bb`` and ``.bbappend`` files in under ``layers/<layer-dependency>/``
+are only included in ``BBPATH`` if the dependency is resolved, i.e. the layer is presant. An example
+of this can be found in meta-bistro_.
```

Comment:
Maybe give a reference to the exact file where this example can be seen in meta-bistro?